AY2021S1-CS2103-T16-3/tp
# Update Developer Guide with Model component and semester archival

### What this does
This PR documents the semester archival feature and adds the use case for archival. It also updates the Model component to reflect the architecture of ResiReg (it was still stuck on AddressBook)


Closes #133 .

### How to test

1. Navigate to the docs folder.
2. Run `bundle exec jekyll serve`.
3. Go to `localhost:4000`.
4. Read through UC10, as well as the archive command feature. See that the diagrams follow their notation, and feel free to nitpick my grammar as well.
5. Nitpick my Model component.

### Notes

While working on this documentation, I noticed a magic string literal "archive.json". I extracted that out into a constant.
